src/Descriptions/Core.idr

```diff
@@ -95,11 +95,11 @@ Untag {e} d = Arg CtorLabel (\l => Arg (Tag l e) (\t => d l t))
 ||| The call will almost always look like `Synthesize d (Data d)`.
 |||
 ||| @ d the description
-||| @ X the translated description, to "tie the recursive knot"
-Synthesize : {Ix: _} -> (d : Desc Ix) -> (X : Ix -> Type) -> (Ix -> Type)
-Synthesize (Ret ix)       X jx = ix ~=~ jx
-Synthesize (Arg A kdesc)  X jx = (a: A ** Synthesize (kdesc a) X jx)
-Synthesize (Rec ix kdesc) X jx = (x: X ix ** Synthesize kdesc X jx)
+||| @ f the translated description, to "tie the recursive knot"
+Synthesize : {Ix: _} -> (d : Desc Ix) -> (f : Ix -> Type) -> (Ix -> Type)
+Synthesize (Ret ix)       f jx = ix ~=~ jx
```

Comment:
It would be nice if we could use new names that were consistent with the old names like before. I associate `f` and `g` too much with functions rather than types.

I would suggest the following renaming scheme `Foo` (old capital)  to `fooTy` (small letters ending with `t`). This is similarly to what you did with `Interface` to `interfaceTy` below.